extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -77,19 +78,43 @@ class AmpInstagram extends AMP.BaseElement {
   }
 
   /** @override */
-  prerenderAllowed() {
-    return true;
+  createPlaceholderCallback() {
+    const placeholder = this.getWin().document.createElement('div');
+    placeholder.setAttribute('placeholder', '');
+    const image = this.getWin().document.createElement('amp-img');
+    // This will redirect to the image URL. By experimentation this is
+    // always the same URL that is actually used inside of the embed.
+    image.setAttribute('src', 'https://www.instagram.com/p/' +
+        encodeURIComponent(this.shortcode_) + '/media/?size=l');
+    image.setAttribute('width', this.element.getAttribute('width'));
+    image.setAttribute('height', this.element.getAttribute('height'));
+    image.setAttribute('layout', 'responsive');
```

Comment:
Dropped `width` and `height` and switched to `layout=fill`, also dropped `position:absolute` in the `setStyles` since `layout=fill` will have absolute.